### ISSUE TITLE
Suggestion - add the ability to append 'attributes' to CanvasPositionMarker and CanvasPath

### ISSUE THREAD

**Issue by svolosd** [TomSchimansky/TkinterMapView#11]
Hi Tom,

It would be useful to provide the ability to add an attributes object to the marker object so that when it is returned to the marker_click(marker) command, the marker object carries with it this attributes object.

I have called it 'attributes' in the modified constructor for the CanvasPositionMarker object.

```python
class CanvasPositionMarker
    def __init__(self, map_widget: "TkinterMapView", position, text=None, attributes=None, text_color="#652A22", font=None,
                 marker_color_circle="#9B261E", marker_color_outside="#C5542D", command=None, image=None,
                 image_zoom_visibility=(13, float("inf"))):
        self.map_widget = map_widget
        self.position = position
        self.text_color = text_color
        self.marker_color_circle = marker_color_circle
        self.marker_color_outside = marker_color_outside
        self.text = text
        self.image = image
        self.image_hidden = False
        self.image_zoom_visibility = image_zoom_visibility
        self.connection_list = []
        self.deleted = False
        self.command = command

        self.polygon = None
        self.big_circle = None
        self.canvas_text = None
        self.canvas_image = None
        self.attributes = None  #added line here.

        if font is None:
            if sys.platform == "darwin":
                self.font = "Tahoma 13 bold"
            else:
                self.font = "Tahoma 11 bold"
        else:
            self.font = font

```
I suggest modifying the CanvasPath in the same way would also be extremely useful as it would provide for having this data within the command on path_click event.

I am adding an attributes dictionary to the marker object as per the modified constructor above in my test app and having this data available within the marker_click handler method is pretty valuable.

Just a suggestion!

Thanks again for taking the time to build this awesome little library.

David.

**Comment by svolosd** [comment]
Hi Tom,

Slight error in the example above.
Should be:


class CanvasPositionMarker
    def __init__(self, map_widget: "TkinterMapView", position, text=None, attributes=None, text_color="#652A22", font=None,
                 marker_color_circle="#9B261E", marker_color_outside="#C5542D", command=None, image=None,
                 image_zoom_visibility=(13, float("inf"))):
        self.map_widget = map_widget
        self.position = position
        self.text_color = text_color
        self.marker_color_circle = marker_color_circle
        self.marker_color_outside = marker_color_outside
        self.text = text
        self.image = image
        self.image_hidden = False
        self.image_zoom_visibility = image_zoom_visibility
        self.connection_list = []
        self.deleted = False
        self.command = command

        self.polygon = None
        self.big_circle = None
        self.canvas_text = None
        self.canvas_image = None
        self.attributes = attributes  #added line here.

        if font is None:
            if sys.platform == "darwin":
                self.font = "Tahoma 13 bold"
            else:
                self.font = "Tahoma 11 bold"
        else:
            self.font = font

David

**Comment by TomSchimansky** [comment]
Hi, I added a `data` attribute now for `CanvasPositionMarker`, `CanvasPath` and `CanvasPolygon`. You can pass it to the objects right when you create them and access or edit it with the `data` attribute like you intended to:
```python
marker_3 = map_widget.set_marker(52.55, 13.4, text="52.55, 13.4", data="some_data_object")

data = marker_3.data
``` 
Thee `data` attribute is `None` by default, it's available now with version 3.10.

**Comment by svolosd** [comment]
Hi Tom,

Thanks for making this change, looks great!

David
